core/scripts/chaincli/handler/debug.go

```diff
@@ -138,8 +140,22 @@ func (k *Keeper) Debug(ctx context.Context, args []string) {
 	// check upkeep
 	if triggerType == ConditionTrigger {
 		message("upkeep identified as conditional trigger")
+
+		if len(args) > 1 {
+			// if a block number is provided, use that block for both checkUpkeep and simulatePerformUpkeep
+			var blockNumber int64
+			blockNumber, err = strconv.ParseInt(args[1], 10, 64)
```

Comment:
should we re-use `blockNum` that is declared on line 135? I believe that is the value that is used in the streams lookup section. If the users passes a blocknum for conditional upkeeps, we should use that block num for stream lookup callbacks, yeah?

Reply:
They are different in my opinion. 
1. The blockNumber in L147 is from user input, and it is used only for conditional upkeeps. So users can run checkUpkeep and performUpkeep using that specific block number.
2. The blockNumber in L135 is used in log trigger based upkeeps, and the block number is read from the transaction receipt. For log trigger upkeeps, we dont ask for block number. we have the tx hash tho.

Let me know if the above makes sense :)

Reply:
Updated~ It is a good idea to reuse the variable, hence removed the additional variable :)